content/blog/2022-03-31-stale-models.md

```diff
@@ -0,0 +1,223 @@
+share models, data, and results can save your team a lot of time and
+frustration.
+
+## Set up data drift reports
+
+When the data in production starts to look different from the data that your
+model was trained, this is called data drift. There are a number of tools that
+help monitor for data drift like [evidently.ai](https://docs.evidentlyai.com/)
+or [Aporia](https://docs.aporia.com/).
+
+Since we're working with Evidently.ai, you can see target drift report when you
+run the notebook for the initial project they made. Here's what it looks like.
+
+![image of the report showing the target drift](https://thumb.tildacdn.com/tild6336-3231-4736-b136-646539326135/-/format/webp/4_week3_pred_actual.png)
+
+So we see at the end of Week 3 the model is in pretty bad shape. This is where
+we can bring in DVC to help us get this stale model off of production faster.
+
+## Running a training experiment to get production up to date
+
+We'll start by taking a year's worth of data and creating a new model. This
+might give us a more accurate model to push to production than using weekly
+data. So we'll take all of the data from 2011 and make our training and testing
+datasets. Then we'll check this data into DVC so it can version it with the
+following commands:
+
+```dvc
+$ dvc add data/train.pkl data/test.pkl
+$ git add data/.gitignore data/train.pkl.dvc data/test.pkl.dvc
+```
+
+We add the `.dvc` files to Git to ensure that we are only checking in the
+metadata for the datasets and not the entire dataset files. Now we can run the
+entire MLOps pipeline with this command:
+
+```dvc
+$ dvc exp run
+```
+
+This will execute the commands we've defined in `dvc.yaml` and it will give us
+the metrics to evaluate how good the model is. Let's take a look at the metrics
+so far with the following command:
+
+```dvc
+$ dvc exp show --no-timestamp
+```
+
+```dvctable
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+┃ neutral:**Experiment**              ┃ metric:**avg_prec** ┃ metric:**roc_auc** ┃ param:**train.seed** ┃ param:**train.n_est** ┃ param:**train.min_split** ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+│ workspace               │  0.70164 │ 0.51384 │ 20210428   │ 450         │ 64              │
+│ main                    │  0.60791 │ 0.45758 │ 20210428   │ 375         │ 64              │
+│ └── 801fdff [exp-a80c0] │  0.70164 │ 0.51384 │ 20210428   │ 450         │ 64              │
+└─────────────────────────┴──────────┴─────────┴────────────┴─────────────┴─────────────────┘
+```
+
+This model doesn't have the best metrics, so we can run more experiments to see
+if tuning hyperparameters will help before we deploy this model to production.
+Let's change the values of the `train.n_est` and `train.n_est` hyperparameters.
+We'll run several experiments with different values and it will produce a table
```

Comment:
Link to https://dvc.org/doc/user-guide/experiment-management and/or explain a bit more what we're doing here.

Reply:
I linked to the user guide since experiments are a secondary thing to the post.